test/test_partition.cpp

```diff
@@ -354,6 +354,17 @@ TYPED_TEST(PartitionIntegerTests, TestPartition)
             typename thrust::device_vector<T>::iterator d_iter
                 = thrust::partition(d_data.begin(), d_data.end(), is_even<T>());
 
+            size_t h_iter_size1 = static_cast<size_t>(thrust::distance(h_data.begin(), h_iter));
```

Comment:
I think we can avoid this with a simpler solution. If we add 
```
        if(count == 0)
            return;
```
to the following lines:
https://github.com/ROCmSoftwarePlatform/rocThrust/blob/develop/thrust/system/hip/detail/sort.h#L120
https://github.com/ROCmSoftwarePlatform/rocThrust/blob/develop/thrust/system/hip/detail/sort.h#L283

I can't push to the repository currently. See my email.